docs/pages/guides/localization.mdx

```diff
@@ -0,0 +1,385 @@
+};
+const i18n = new I18n(translations);
+
+// Set the locale once at the beginning of your app.
+i18n.locale = Localization.locale;
+
+// When a value is missing from a language it'll fall back to another language with the key present.
+i18n.enableFallback = true;
+// To see the fallback mechanism uncomment the line below to force the app to use the Japanese language.
+// i18n.locale = 'ja';
+
+export default function App() {
+  return (
+    <View style={styles.container}>
+      <Text style={styles.text}>
+        {i18n.t('welcome')} {i18n.t('name')}
+      </Text>
+      <Text>Current locale: {i18n.locale}</Text>
+      <Text>Device locale: {Localization.locale}</Text>
+    </View>
+  );
+}
+
+/* @hide const styles = StyleSheet.create({ ... }); */
+const styles = StyleSheet.create({
+  container: {
+    backgroundColor: '#fff',
+    alignItems: 'center',
+    justifyContent: 'center',
+    flex: 1,
+  },
+  text: {
+    fontSize: 20,
+    marginBottom: 16,
+  },
+});
+/* @end */
+```
+
+</SnackInline>
+
+
+<Collapsible summary="Using other translation libraries">
+
+
+
+While this guide uses `i18n-js` as an example, there are many other libraries that can help you with translations, and they each have different features:
+
+- Creating translations is a large effort that you'd ideally want to hire someone for, but it's hard for them to edit translations directly in your code. Some translation libraries can integrate with translation management tools (essentially a web service that lets you outsource, auto-generate or just makes it easier to create translations).
+
+- Some libraries make it easy to move the component structure around (and not just words of your copy) based on translation strings. Imagine you want to localize a string that includes a `<Pressable>` link, and you want the link to be in a different order from the rest of the text depending on the translation.
+
+Here are some libraries that can be a good option depending on your needs:
+  
+- [`React i18next`](https://react.i18next.com/)
+
+  `React i18next` is a stable, well-maintained library based on `i18next`.
+
+- [`typesafe-i18n`](https://github.com/ivanhofer/typesafe-i18n)
+
+  `typesafe-i18n` compiles translations so that it saves space in the production bundle. It also generates typescript types for your translations, so that you can use them in your code and get autocomplete and type safety.
+  It requires the `Intl` API for some features, so it's only usable in projects with Hermes enabled.
+
+</Collapsible>
+
+### Translating app metadata
+
+If you plan on shipping your app to different countries, or regions, or want it to support various languages, you can provide [localized](/versions/latest/sdk/localization) strings for things like the display name and system dialogs. All of this is easily set up [in your `app.json`](/workflow/configuration) file. First, set `ios.infoPlist.CFBundleAllowMixedLocalizations: true`, then provide a list of file paths to `locales`.
```

Comment:
```suggestion
If you plan on shipping your app to different countries or regions or want it to support various languages, you can provide [localized](/versions/latest/sdk/localization) strings for things like the display name and system dialogs. This is easily set up [in the Expo config](/workflow/configuration) file. First, set `ios.infoPlist.CFBundleAllowMixedLocalizations: true`, then provide a list of file paths to `locales`.
```

[As per our writing style guide, file names should be bold text.](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md#referencing-android-ios-and-web)